src/vscode-dts/vscode.proposed.badges.d.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/62783 @matthewjamesadam
+
+	/**
+	 * Base class for badges
+	 */
+	export abstract class Badge {
+		/**
+		 * A label to present in tooltips
+		 */
+		readonly label: string;
+	}
+
+	/**
+	 * A badge presenting a number
+	 */
+	export class NumberBadge extends Badge {
+
+		/**
+		 * Creates a new number badge.
+		 *
+		 * @param number The number to present
+		 * @param label The label to present in tooltips
+		 */
+		constructor(number: number, label: string);
+
+		/**
+		 * The number to present
+		 */
+		readonly number: number;
```

Comment:
Instead of calling this `number`, we can call it `value`, still with type `number`. If we decide to support non-number badges later the type can be an or-type.

Reply:
Done.